src/components/SubmitTokenInfo/index.tsx

```diff
@@ -0,0 +1,234 @@
+import { utils } from '@ckb-lumos/base'
+import CloseIcon from '../../assets/modal_close.png'
+import HelpIcon from '../../assets/qa_help.png'
+import AlertIcon from '../../assets/alert.png'
+import { LabeledInput } from './LabeledInput'
+import styles from './styles.module.scss'
+import { ImgUpload } from './ImgUpload'
+import CommonButton from '../CommonButton'
+import CommonModal from '../CommonModal'
+import { submitTokenInfo } from '../../services/ExplorerService/fetcher'
+import CommonSelect from '../CommonSelect'
+import { isMainnet } from '../../utils/chain'
+import { MainnetContractHashTags, TestnetContractHashTags } from '../../constants/scripts'
+import { isValidPositiveInteger } from '../../utils/number'
+
+type Props = {
+  isOpen: boolean
+  onClose: () => void
+}
+
+const LabelTooltip = ({ title, icon }: { title: string; icon?: string }) => (
+  <Tooltip placement="bottom" title={title}>
+    <img src={icon ?? HelpIcon} alt="tooltip" className={styles.tooltipIcon} />
+  </Tooltip>
+)
+
+export const SubmitTokenInfo = ({ onClose, isOpen }: Props) => {
+  const { t } = useTranslation()
+  const [args, setArgs] = useState('')
+
+  const [symbol, setSymbol] = useState('')
+  const [name, setName] = useState('')
+  const [decimal, setDecimal] = useState('')
+  const [description, setDescription] = useState('')
+  const [website, setWebsite] = useState('')
+  const [creatorEmail, setCreatorEmail] = useState('')
+  const [logo, setLogo] = useState<string | null>(null)
+
+  const scriptDataList = isMainnet() ? MainnetContractHashTags : TestnetContractHashTags
+  const tokenTypeOptions = scriptDataList
+    .filter(scriptData => scriptData.tag.includes('sudt'))
+    .sort((a, b) => a.tag.localeCompare(b.tag))
+    .map(scriptData => ({
+      value: scriptData.codeHashes[0],
+      label: scriptData.tag,
+    }))
+  const [codeHash, setCodeHash] = useState<string>(tokenTypeOptions[0].value)
+  const handleTokenTypesChange = (value: string) => {
+    setCodeHash(value)
+  }
+
+  useEffect(() => {
+    // scroll to top when open modal
+    if (isOpen) {
+      window.scrollTo(0, 0)
+    }
+  }, [isOpen])
+
+  const handleImgChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0]
+    if (file) {
+      const reader = new FileReader()
+      let baseString: string | null = null
+      // eslint-disable-next-line func-names
+      reader.onloadend = function () {
+        baseString = reader.result as string
+        setLogo(baseString)
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const clearForm = () => {
+    onClose()
+    setArgs('')
+    setSymbol('')
+    setName('')
+    setDecimal('')
+    setDescription('')
+    setWebsite('')
+    setCreatorEmail('')
+    setLogo(null)
+  }
+
+  const handleClose = () => {
+    onClose()
+  }
+
+  const validateFields = () =>
+    isValidPositiveInteger(decimal) && !codeHash && !symbol && !args && !website && !creatorEmail
```

Comment:
`isValidPositiveInteger(decimal` returns `true` if `decimal` meets the requirement
`!codeHash` returns `false` if `codeHash` meets the requirement
So do `symbol`, `args`, `website` and `creatorEmail`

I'm confused.

Reply:
And `decimal to be 0` is allowed, so `isValidPositiveInteger` should be `isValidNonNegativeInteger`, as mentioned at https://github.com/Magickbase/ckb-explorer-frontend/pull/127/#discussion_r1445715905

Reply:
![image](https://github.com/Magickbase/ckb-explorer-frontend/assets/7271329/fd35fe7d-3dae-4850-af8b-963f61b30a53)

`Confirm` button is disabled when all fields are valid

Reply:
updated by [5859a54](https://github.com/Magickbase/ckb-explorer-frontend/pull/127/commits/5859a546e4068b359c00e002f856d68cdf8849e4)